src/api.v2/helpers/access/createUserInvite.js

```diff
@@ -0,0 +1,40 @@
+const UserAccess = require('../../model/UserAccess');
+
+const { getAwsUserAttributesByEmail } = require('../../../utils/aws/user');
+const sendEmail = require('../../../utils/send-email');
+const buildUserInvitedEmailBody = require('../../../utils/emailTemplates/buildUserInvitedEmailBody');
+const buildUserInvitedNotRegisteredEmailBody = require('../../../utils/emailTemplates/buildUserInvitedNotRegisteredEmailBody');
+
+const OK = require('../../../utils/responses/OK');
+
+const getLogger = require('../../../utils/getLogger');
+
+const logger = getLogger('[AccessModel] - ');
+
+const createUserInvite = async (experimentId, userEmail, role, inviterUser) => {
+  let userAttributes;
+  let emailBody;
+
+  try {
+    userAttributes = await getAwsUserAttributesByEmail(userEmail);
+
+    const userSub = userAttributes.find((attr) => attr.Name === 'sub').Value;
+    new UserAccess().grantAccess(userSub, experimentId, role);
+    emailBody = buildUserInvitedEmailBody(userEmail, experimentId, inviterUser);
+  } catch (e) {
+    if (e.code !== 'UserNotFoundException') {
+      throw e;
+    }
+
+    logger.log('User has not yet signed up, inviting new user');
```

Comment:
the meaning of the sentence for this log is hard to understand, at least for me. Can you change to "This user doesn't have an account yet. They will be invited as a new user."

Reply:
Can you assume that all exceptions that are not `UserNotFoundException` are due to the user not having an account? It will be safer if you can check explicitly if the error is due to the user not having an account and then execute these lines of code, rather than assume that all possible catched errors are this.

Reply:
We ccan be sure its UserNotFoundException here because if its not it would have been thrown on line 29

Reply:
Changed to: 'Invited user does not have an account yet. Sending invitation email.'

Reply:
line 26 throws only if the code is `UserNotFoundException`. If the error is anything else, we will always be creating new user.

Reply:
line 26 will throw if the error is not `UserNotFoundException`

Reply:
oh sorry I can't read if statements with ! in them. Looks good then